provisioning/device/test/_symmetric_key_registration_test.js

```diff
@@ -0,0 +1,147 @@
+      var transport = {
+        setSasToken: sinon.spy(),
+      };
+      var fakeError = new Error('got a token create Error');
+      var security = {
+        createAuthenticationToken: sinon.stub().callsArgWith(1, fakeError),
+        getRegistrationId: sinon.stub().callsArgWith(0, null, fakeRegistrationId)
+      };
+      var clientObj = new SymmetricKeyRegistration(fakeProvisioningHost, fakeIdScope, transport, security);
+      clientObj.register(function(err, response) {
+        assert(security.createAuthenticationToken.calledOnce);
+        assert.isOk(err);
+        assert.strictEqual(fakeError, err);
+        callback();
+      });
+    });
+
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_010: [ if the polling register returns an error, the `register` shall invoke the `_callback` with that error. ] */
+    it('fails if register fails', function(callback) {
+      var transport = {
+        setSasToken: sinon.spy(),
+      };
+      var security = {
+        createAuthenticationToken: sinon.stub().callsArgWith(1, null, fakeSasToken),
+        getRegistrationId: sinon.stub().callsArgWith(0, null, fakeRegistrationId)
+      };
+      var fakeError = new Error('got a register Error');
+      var clientObj = new SymmetricKeyRegistration(fakeProvisioningHost, fakeIdScope, transport, security);
+      clientObj._pollingStateMachine.register = sinon.stub().callsArgWith(1, fakeError);
+      clientObj._pollingStateMachine.disconnect = sinon.stub().callsArgWith(0,null);
+      clientObj.register(function(err, response) {
+        assert.isOk(err);
+        assert.strictEqual(fakeError, err);
+        callback();
+      });
+    });
+
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_010: [ if the polling register returns an error, the `register` shall invoke the `_callback` with that error. ] */
+    it('disconnect failure does NOT affect results', function(callback) {
+      var transport = {
+        setSasToken: sinon.spy(),
+      };
+      var security = {
+        createAuthenticationToken: sinon.stub().callsArgWith(1, null, fakeSasToken),
+        getRegistrationId: sinon.stub().callsArgWith(0, null, fakeRegistrationId)
+      };
+      var fakeError = new Error('got a disconnect Error');
+      var clientObj = new SymmetricKeyRegistration(fakeProvisioningHost, fakeIdScope, transport, security);
+      clientObj._pollingStateMachine.register = sinon.stub().callsArgWith(1, null, { registrationState: fakeResponse } );
+      clientObj._pollingStateMachine.disconnect = sinon.stub().callsArgWith(0,fakeError);
+      clientObj.register(function(err, response) {
+        assert.isNotOk(err);
+        assert(clientObj._pollingStateMachine.disconnect.calledOnce);
+        callback();
+      });
+    });
+  });
+
+  describe('#cancel', function () {
+    /* Tests_SRS_NODE_DPS_X509_REGISTRATION_18_003: [ `cancel` shall call `cancel` on the transport object. ] */
+    it('calls cancel on the transport', function(callback) {
```

Comment:
>cancel [](start = 14, length = 6)

what if cancel fails? can it fail?

Reply:
This is exactly what is tested for tpm and x509.

---
In reply to: [231362141](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231362141) [](ancestors = 231362141)

Reply:
.... and? we shipped incomplete stuff in the past so let's keep making incomplete stuff? I don't think there's time pressure here. let's do this right and test our code paths properly.

---
In reply to: [231621585](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231621585) [](ancestors = 231621585,231362141)

Reply:
Done.

---
In reply to: [232119503](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r232119503) [](ancestors = 232119503,231621585,231362141)